frontend/src/ReportDataHandler.js

```diff
@@ -0,0 +1,18 @@
+import { useState, useEffect } from "react";
+import { useLoaderData, useNavigation } from "react-router-dom";
+
+import App from "./App";
+import LoadingAnimation from "./components/LoadingAnimation";
+
+const ReportDataHandler = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const loaderData = useLoaderData();
+
+  useEffect(() => {
+    setIsLoading(!isLoading);
+  }, [useNavigation().state === "loading"]);
```

Comment:
In this case, `useNavigation` will only change its state when there is a path change. And for that case, we will require to trigger the useEffect. So I don't think there any problem here.

Reply:
But there's no need to use `useEffect`, you just need simple variable assignment:
```js
const isLoading = navigationData.state === 'loading'
```
If simplified it could look like this:
```js
const ReportDataHandler = () => {
  const loaderData = useLoaderData()
  const {state} = useNavigation()

  return state === 'loading' ? <LoadingAnimation /> : <App lines={loaderData} />
}
```

Reply:
Hmm! got it!